warp10/src/main/java/io/warp10/script/WarpScriptSingleValueFillerFunction.java

```diff
@@ -23,10 +23,20 @@ public interface WarpScriptSingleValueFillerFunction {
   public interface Precomputable extends WarpScriptSingleValueFillerFunction {
     public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException;
 
-    default public Object evaluate(long tick) throws WarpScriptException {
+    default public void fillTick(long tick, GeoTimeSerie gts, Object invalidValue) throws WarpScriptException {
       throw new WarpScriptException("Invalid Filler Error: evaluator function has not been precomputed yet.");
     }
+
+    @Deprecated
+    default public Object evaluate(long tick) throws WarpScriptException {
+      throw new WarpScriptException("Invalid Filler Definition Error: evaluate function is deprecated.");
+    }
   }
 
-  public Object evaluate(long tick) throws WarpScriptException;
+  public void fillTick(long tick, GeoTimeSerie gts, Object invalidValue) throws WarpScriptException;
```

Comment:
Do you mean to modify the gts inplace ?
This is also a breaking change

Reply:
Nvm, you pass a clone of the input gts when you use fillTick in fill.
But then I dont understand the point. Why do you need fillTick rather than evaluate ?
Why implementing the new precomputable interface with evaluate isn't enough to make older filler faster ?

Reply:
> Do you mean to modify the gts inplace ? This is also a breaking change

Well, no... gts is cloned very early in the process. Please provide an example of malfunction.

Reply:
I kept evaluate for retro compatibility, and for filler.trend (the last slow one, and broken imho).

Main reasons:
- support geo
- the gts argument allow to call setvalue directly in the filler, instead of returning values to the caller (including geo).